node/node.go

```diff
@@ -157,9 +157,10 @@ type Node struct {
 	// services
 	eventBus         *types.EventBus // pub/sub for services
 	stateDB          dbm.DB
-	blockStore       *bc.BlockStore         // store the blockchain to disk
-	bcReactor        *bc.BlockchainReactor  // for fast-syncing
-	mempoolReactor   *mempl.MempoolReactor  // for gossipping transactions
+	blockStore       *bc.BlockStore        // store the blockchain to disk
+	bcReactor        *bc.BlockchainReactor // for fast-syncing
+	mempoolReactor   *mempl.Reactor        // for gossipping transactions
+	mempool          *mempl.CListMempool
```

Comment:
Why is this a pointer while in `rpc/core/pipe.go` it was changed to not be a pointer?

Reply:
Because node's mempool calls InitWAL/CloseWAL, but I don't want to add these methods to Mempool interface, which is already big

Reply:
I created `mempoolWithWAL` interface

Reply:
Made them a part of Mempool interface for now.

1. WAL is an implementation details and shouldn't really be a part of the interface.
2. But we need a way to init/close associated resources
3. And we need a way to configure them (wal path, etc.)

so we should either make node.Mempool a concrete type OR make InitWAL/CloseWAL/ConfigureWAL a part of any mempool interface